packages/block-editor/src/components/link-control/link-preview.js

```diff
@@ -7,49 +7,99 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Button, ExternalLink } from '@wordpress/components';
+import {
+	Button,
+	ExternalLink,
+	__experimentalText as Text,
+} from '@wordpress/components';
 import { filterURLForDisplay, safeDecodeURI } from '@wordpress/url';
+import { Icon, globe } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import { ViewerSlot } from './viewer-slot';
 
+import useRemoteUrlData from './use-remote-url-data';
+
 export default function LinkPreview( { value, onEditClick } ) {
+	const { richData, isFetching } = useRemoteUrlData( value?.url );
+
+	// Rich data may be an empty object so test for that.
+	const hasRichData = richData && Object.keys( richData ).length;
+
 	const displayURL =
 		( value && filterURLForDisplay( safeDecodeURI( value.url ), 16 ) ) ||
 		'';
-
 	return (
 		<div
 			aria-label={ __( 'Currently selected' ) }
 			aria-selected="true"
 			className={ classnames( 'block-editor-link-control__search-item', {
 				'is-current': true,
+				'is-rich': hasRichData,
+				'is-fetching': !! isFetching,
+				'is-preview': true,
```

Comment:
Is `has-preview` more suitable here?

Reply:
Actually not in this case as the "preview" is not related to the rich data preview but rather the "preview" of the link in general.

So the status is correct this _is_ a _preview_ rather than it _has_ a _preview_.